### DISCUSSION TITLE
Embeddable Widget Code needed for Player to show Metadata during LIVESTREAMING

### DISCUSSION THREAD

**Discussion by JeffersonStarseed** [AzuraCast/AzuraCast#7156] [Q&A] [answered]
I don't know if anyone else is having this problem, but none of the Azuracast stations that I have set up are showing or displaying "artist & song" information (metadata) during a DJ's "LIVESTREAM".  All it says or displays is "LIVE", and that's it.  Regular AutoDJ (Liquidsoap) has no problem with displaying artist/song/album when playing recorded songs uploaded into the AutoDJ rotation, but it will not show anything when a DJ is actually doing a "Livestream" other than the word "LIVE".  I used the player widget-code provided by Azuracast to embed on the DJ's webpages, but I get the same results whether its embedded on webpages or ...if I just launch Azuracasts own internal public station page.   What am I doing wrong?

When we did our live streams to Caster.fm or Zeno.fm, we never had any problem.  It always showed (at least) the artist and the song title when we live-streamed to our stations with those platform-providers.  All us DJ's have the proper encoder-metadata set in our streaming software to send-out to Azuracast, just like when we used Caster or Zeno.  I use Virtual DJ, and some of the others are using MIXXX.

These (below) are examples of players from Caster showing normal streamed metadata:
![newplayer1](https://github.com/AzuraCast/AzuraCast/assets/159277623/3cd83b15-ac0f-4d6f-a8c5-66a4c52f62e8)
     
![userpage_dark_sc](https://github.com/AzuraCast/AzuraCast/assets/159277623/7d051f07-2f6c-4627-b934-b9ca5f995edf)

I know this is possible with Azuracast, because I used to Live-stream a show called the "Saturday Night Special" to Radiomina.com, and their radio platform is a standard Azuracast server, and it showed the artist and song title of whatever I was live-streaming to it with my Virtual DJ software back then.

**_How can I make "artist" and "song title" appear on these Azuracast stations I've set up when the DJ's are Live doing their "Live-streams" from their broadcasting software?_**  What am I missing here?  What am I doing wrong?

**Comment by Heimatradio** [comment]
If 'Live Broadcast' is displayed, no ID tags are received from the streamer.

**Comment by JeffersonStarseed** [comment]
That's what I'm talking about.  When I live-stream to Caster.fm or Zeno.fm, ...it shows the metadata.  When I live-stream to Radiomina.com (an Azuracast server), ...it shows the metadata.  The only Azuracast servers that it does 'not' show the artist/song on when I live-stream are the Azuracast servers I have created/made/installed.  I'm trying to figure-out what I've done wrong, or what settings I've set wrong, in Azuracast.  My metadata/tags are set-up to transmit properly from Virtual DJ, and always have been.
I double-checked the players.  My apologies ...they say "LIVE" (not 'Live Broadcast').  Even though in the settings it's set up to display 'Live Broadcast' when there is no metedata being received, that's not what comes up.  It just says "LIVE".

**Reply by LordHelmchen666** [reply]
What's in Liquidsoap logs when a DJ connects?  There should be some informations what happens.

**Comment by JeffersonStarseed** [comment, marked as answer]
UPDATE:  I meant to give this update many months ago earlier in the year. My apologies.  This of course was not an issue with Azuracast, but several common issues outside of it, the first being that metadata is "different" for different kinds of audio files (ex: ogg metadata formatting is not the same as MP3). Some of us assume that metadata is automatic, or automatically written or adjusted for the different types of audio files by DJ software or trancoding, but not always, depending on what you're using, ...and the other culprit is settings and "format fields" in the broadcasting software or encoder software.

For our purposes, we have found it easier for those DJ's involved to keep files streamed (from their DJ broadcasting software to Azuracast) at "high-kbps" MP3 (such as 256 or 320), and have Azuracast set to stream-out to the listener over a better, but lower kbs format (such as AAC+ or Ogg Opus).

By keeping the initial stream "to" Azuracast at MP3 format, the metadata and session info (in any format-fields) is received properly by Azuracast and properly displayed on the radio player "during a live broadcast".

If choosing to stream to Azuracast using an Ogg or Opus streaming format (for example), one must make sure that file metadata is in the proper format.  If not, it will need to be changed/re-tagged/reformatted to an Ogg or Opus file format metadata.

A NOTE: (for Virtual DJ users)
In the case of "Virtual DJ" broadcasting software, it has it's own limitations and issues.  Several of the fields under general "Broadcasting" are dead or null (such as the 'Session Name' and "Other Fields'), and we have already addressed VDJ with this issue.

However, we have found a workaround for this issue within its core "SongInfoFormat" field under "OPTIONS: Broadcast". Please see this pictorial post on the Virtual DJ discussion forum for the workaround: 
https://virtualdj.com/forums/259373/VirtualDJ_Technical_Support/Broadcast_-_Other_Fields_.html
